libc/string_cmp.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - string.h
+ * TESTED:
+ *    - memcmp()
+ *    - strcmp()
+ *    - strncmp()
+ *    - strcoll()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Damian Modzelewski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <limits.h>
+#include <unity_fixture.h>
+
+
+#define BUFF_SIZE 128
+
+
+TEST_GROUP(string_cmp);
+
+
+TEST_SETUP(string_cmp)
+{
+}
+
+
+TEST_TEAR_DOWN(string_cmp)
+{
+}
+
+
+TEST(string_cmp, cmp)
```

Comment:
Only a quick look: please split this test to a few cases and make sure that there are no more edge cases/things to check.